odoo/tools/_monkeypatches.py

```diff
@@ -65,3 +69,8 @@ def literal_eval(expr):
     return orig_literal_eval(expr)
 
 ast.literal_eval = literal_eval
+
+if MIN_PY_VERSION >= (3, 12):
+    warnings.warn(f"You need to remove the monkey patching for `num2words` library as Odoo requires Python >= {'.'.join(map(str, MIN_PY_VERSION))} and it's already fixed from version {'.'.join(map(str, MIN_PY_VERSION))}", DeprecationWarning, stacklevel=1)
```

Comment:
I think the part about the version is not really useful, and might even be wrong (the "fixed from" version number should be hardcoded). I also think it might be useful to remind to bump the version of the library.
Rewording suggestion:
```suggestion
    warnings.warn("The num2words monkey patch is obsolete. Bump the version of the library to the latest available in the official package repository, if it hasn't already been done, and remove the patch.", DeprecationWarning, stacklevel=1)
```

Reply:
I don't think DeprecationWarning is relevant here. And I'm not sure how the warning should be logged. It seems that another patch in the same file uses `_logger.error`; maybe that's the standard way to go?